tempest/api/workloadmgr/restore/test_image_booted.py

```diff
@@ -1,323 +1,357 @@
+import tempest
+import unicodedata
+import collections
+sys.path.append(os.getcwd())
+from tempest.api.workloadmgr import base
+from tempest import config
+from tempest import test
+from oslo_log import log as logging
+from tempest import tvaultconf
+from tempest import reporting
+import time
+from tempest import command_argument_string
+from tempest.util import cli_parser
+from tempest.util import query_data
+
+LOG = logging.getLogger(__name__)
+CONF = config.CONF
+
+class WorkloadTest(base.BaseWorkloadmgrTest):
+
+    credentials = ['primary']
+
+    @classmethod
+    def setup_clients(cls):
+        super(WorkloadTest, cls).setup_clients()
+        cls.client = cls.os.wlm_client
+    
+    def assign_floating_ips(self, vm_id, cleanup):
+        fip = self.get_floating_ips()
+        LOG.debug("\nAvailable floating ips are {}: \n".format(fip))
+        self.set_floating_ip(str(fip[0]),vm_id,floatingip_cleanup=cleanup)
+        return(fip[0])        
+
+    def data_ops(self, flo_ip, data_dir_path, file_count):
+        ssh = self.SshRemoteMachineConnectionWithRSAKey(str(flo_ip))
+        stdin, stdout, stderr = ssh.exec_command("sudo apt install qemu-guest-agent")
+        self.addCustomfilesOnLinuxVM(ssh, data_dir_path, file_count)
+        ssh.close()
+
+    def calcmd5sum(self, flip, data_dir_path):
+        mdb = {}
+        ssh = self.SshRemoteMachineConnectionWithRSAKey(str(flip))
+        mdb[str(flip)] = self.calculatemmd5checksum(ssh, data_dir_path)
+        ssh.close()
+        return mdb
+
+    def create_snapshot(self, workload_id, is_full=True):
+        if is_full == True:
+            substitution = 'Full'
+        else:
+            substitution = 'Incremental'
+
+        snapshot_id, command_execution, snapshot_execution = self.workload_snapshot_cli(workload_id, is_full=is_full)
+        if command_execution == 'pass':
+            reporting.add_test_step("{} snapshot command execution".format(substitution), tvaultconf.PASS)
+            LOG.debug("Command executed correctly for full snapshot")
+        else:
+            reporting.add_test_step("{} snapshot command execution".format(substitution), tvaultconf.FAIL)
+            raise Exception("Command did not execute correctly for full snapshot")
+
+        if snapshot_execution == 'pass':
+            reporting.add_test_step("{} snapshot".format(substitution), tvaultconf.PASS)
+        else:
+            reporting.add_test_step("{} snapshot".format(substitution), tvaultconf.FAIL)
+            raise Exception("Full snapshot failed")
+        return(snapshot_id)
+ 
+ 
+    @test.idempotent_id('9fe07175-912e-49a5-a629-5f52eeada4c9')
+    def test_1_image_booted(self):
+        try:
+            deleted = 0
+            ## VM and Workload ###
+            tests = [['tempest.api.workloadmgr.sanity.test_image_booted_Selective-restore',0], ['tempest.api.workloadmgr.sanity.test_image_booted_Inplace-restore',0], ['tempest.api.workloadmgr.sanity.test_image_booted_Oneclick-restore',0]]
```

Comment:
Replace 'sanity' with 'restore' in test names here

Reply:
Done